src/jdk.jfr/share/classes/jdk/jfr/internal/test/DeprecatedThing.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package jdk.jfr.internal.test;
+
+public class DeprecatedThing {
+    public static int staticCounter;
+    public int counter;
+
+    @Deprecated
+    public void foo() {
+        bar();
+    }
+
+    @Deprecated
+    public void zoo() {
+        System.out.println("Zoo invoked");
+        for (int i = 0; i < 1_000_000; i++) {
+            bar();
+        }
+    }
+
+    private void bar() {
+        baz();
+    }
+
+    public void baz() {
+        counter++;
+    }
+
+    @Deprecated(forRemoval = true)
+    public static void deprecatedForRemovalSlow() {
+        for (int i = 0; i < 1_000_000; i++) {
+            deprecatedForRemovalSlow2();
+        }
+    }
+
+    @Deprecated(forRemoval = true)
+    public static void deprecatedForRemovalSlow2() {
+        staticCounter++;
+    }
+
+    @Deprecated(forRemoval = true)
+    public static void deprecatedForRemovalFast() {
+        staticCounter++;
+    }
+
+    @Deprecated
+    public static void deprecated1() {
+        staticCounter++;
+    }
+
+    @Deprecated(forRemoval = true)
+    public static void deprecated2() {
+        staticCounter++;
+    }
+
+    @Deprecated(forRemoval = true)
+    public static void deprecated3() {
+        staticCounter++;
+    }
+
+    @Deprecated
+    public static void reflection() {
+        staticCounter++;
+    }
+
+    @Deprecated(forRemoval = true)
+    public static void reflectionForRemoval() {
+        staticCounter++;
+    }
```

Comment:
You might want to extend the set of tests to include cases that have the "since" element.  There is a 2x2 matrix of cases to fully exercise the parsing of the RuntimeVisibleAnnotations content.

Reply:
Thanks for this input, Alan.

I have made the tests and cases more structured to cover this. Cheers.